ducktape/tests/runner.py

```diff
@@ -234,13 +250,11 @@ def run_all_tests(self):
                         self._handle(event)
                     except Exception as e:
                         err_str = "Exception receiving message: %s: %s" % (str(type(e)), str(e))
-                        err_str += "\n" + traceback.format_exc(limit=16)
+                        # this is redundant output
+                        # err_str += "\n" + traceback.format_exc(limit=16)
                         self._log(logging.ERROR, err_str)
 
                         # All processes are on the same machine, so treat communication failure as a fatal error
-                        for proc in self._client_procs.values():
-                            proc.terminate()
-                        self._client_procs = {}
```

Comment:
this is what previously propogated the SIGTERM's (if you read the docs thats what multiprocessing terminate does with some more redundancy i believe ).  Anyway this is probably where the new cleanup logic should go

Reply:
Cool. This was my first stab at merging w/ some conflicts. I'll move the hung test logic to the runner and we can see how it looks.